From f54768e436a32bc16d6fc5cdce3118e44d52bcf7 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:34 -0400
Subject: [virt] kvm/x86: remove WARN_ON from get_kernel_ns()

Message-id: <1406040016-3289-180-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86258
O-Subject: [RHEL7 PATCH v2 179/221] KVM: x86: remove WARN_ON from get_kernel_ns()
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Marcelo Tosatti <mtosatti@redhat.com>

Function and callers can be preempted.

https://bugzilla.kernel.org/show_bug.cgi?id=73721

Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit b351c39cc9e0151cee9b8d52a1e714928faabb38)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 6b348e0..fbb43fc 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -1125,7 +1125,6 @@ static inline u64 get_kernel_ns(void)
 {
  struct timespec ts;
 
- WARN_ON(preemptible());
  ktime_get_ts(&ts);
  monotonic_to_bootbased(&ts);
  return timespec_to_ns(&ts);
-- 
1.7.1